modules/consensus/accept.go

```diff
@@ -24,8 +25,40 @@ var (
 	errOrphan                 = errors.New("block has no known parent")
 )
 
-// validHeader does some early, low computation verification on the block.
-func (cs *ConsensusSet) validHeader(tx *bolt.Tx, b types.Block) error {
+type (
+	// dbBucket represents a collection of key/value pairs inside the database.
+	dbBucket interface {
+		Get(key []byte) []byte
+	}
+
+	// dbTx represents a read-only transaction on the database that can be used
+	// for retrieving values.
+	dbTx interface {
+		Bucket(name []byte) dbBucket
+	}
+
+	txBlock interface {
+		ID() types.BlockID
+		GetParentID() types.BlockID
+		CheckMinerPayouts(height types.BlockHeight) bool
+		CheckTarget(target types.Target) bool
+		GetTimestamp() types.Timestamp
+	}
+
+	// boltTxWrapper wraps a bolt.Tx so that it matches the dbBucket interface.
+	boltTxWrapper struct {
+		tx *bolt.Tx
+	}
+)
```

Comment:
Mostly just wanted you to see that all of the functions like 'getSiacoinOutput' are defined in database.go using a more generic 'tx' from boltdb. The interfaces `dbBucket` and `dbTx` should reside in database.go, since they are highly relevant to the database.

I am also a little confused by the `txBlock` interface. `ID` and `CheckMinerPayouts` are both functions operating directly on a block, seems weird that you would mock them. They don't seem like dependencies to me. A block doesn't seem like something you should be mocking.